Ginger/Ginger/AutomatePageLib/AddActionMenu/ApplicationModels/APINavPage.xaml.cs

```diff
@@ -45,18 +45,38 @@ public APINavPage(Context context)
             mContext = context;
 
             AppApiModelsFolderTreeItem mAPIsRoot = new AppApiModelsFolderTreeItem(WorkSpace.Instance.SolutionRepository.GetRepositoryItemRootFolder<ApplicationAPIModel>());
+
             mItemTypeRootNode = mAPIsRoot;
             mAPIPage = new SingleItemTreeViewSelectionPage("API Models", eImageType.APIModel, mItemTypeRootNode, SingleItemTreeViewSelectionPage.eItemSelectionType.Multi, true,
                                         new Tuple<string, string>(nameof(ApplicationAPIModel.TargetApplicationKey) + "." + nameof(ApplicationAPIModel.TargetApplicationKey.ItemName), mContext.Activity.TargetApplication),
                                             UCTreeView.eFilteroperationType.Equals, showAlerts: false) ;
 
             mItemTypeRootNode.SetTools(mAPIPage.xTreeView);
             mAPIPage.xTreeView.SetTopToolBarTools(mAPIsRoot.SaveAllTreeFolderItemsHandler, mAPIsRoot.AddAPIModelFromDocument, RefreshTreeItems);
-
+            mAPIPage.xTreeView.xTreeViewTree.ItemDoubleClick += APIModelItemsTreeView_ItemDoubleClick;
```

Comment:
don't do it on double click, we said it will be added to the API Model tree node context menu.
Do it like this:
1- add Parameter called "AddEditToMenu" to AppApiModelsFolderTreeItem and to AppApiModelTreeItem classes
2- from this class set mAPIsRoot.AddEditToMenu as "True"
3- on AppApiModelsFolderTreeItem pass the AddEditToMenu = True to all child Folders/Tree nodes
4- on AppApiModelTreeItem in SetTools() check the AddEditToMenu flag value and only if it's true then add the Edit context menu option (at first place) which will open the API model edit page